src/data.py

```diff
@@ -0,0 +1,74 @@
+import os
+import cv2
+from typing import Tuple, List
+from sklearn.model_selection import train_test_split
+import pandas as pd
+
+
+def check_images(directory: str) -> Tuple[List[str], int, int]:
+    """
+    Checks the images in the specified directory to identify empty or corrupted files.
+
+    Args:
+        directory (str): The path to the directory containing the images.
+
+    Returns:
+        Tuple[List[str], int, int]:
+            - A list of filenames that are either empty or corrupted.
+            - The count of empty or corrupted images.
+            - The count of correctly loaded images.
+
+    This function iterates through all the files in the given directory, attempting to load each image using OpenCV's `
+    cv2.imread` function.
+    If an image cannot be loaded (i.e., it is empty or corrupted), it is added to the `empty_or_corrupted` list.
+    The function finally returns this list along with the count of corrupted/empty images and the count of successfully
+    loaded images.
+    """
+    empty_or_corrupted: List[str] = []
+    num_correct: int = 0
+
+    for filename in os.listdir(directory):
+        file_path = os.path.join(directory, filename)
+        img = cv2.imread(file_path)
+        if img is None or img.size == 0:
+            empty_or_corrupted.append(filename)
+        else:
+            num_correct += 1
+
+    return empty_or_corrupted, len(empty_or_corrupted), num_correct
```

Comment:
Suggestion: imo the function would be more clear if it just returned empty_or_corrupted list, the lengh of it and the number of correct images are easy to get from it

Reply:
agreed